src/Shared/UnitTests/MockLogger.cs

```diff
@@ -383,6 +398,27 @@ internal void LoggerEventHandler(object sender, BuildEventArgs eventArgs)
             }
         }
 
+        internal void TelemetryEventHandler(object sender, BuildEventArgs eventArgs)
+        {
+            lock (_lockObj)
+            {
+                if (eventArgs is TelemetryEventArgs telemetryEventArgs)
```

Comment:
minor: lot of things happening inside this lock. Please ensure everything locked is a must.

Reply:
Everything in lock scope is a must. We can move lock inside if, but due to fact that if will always be true, from runtime perspective it would not make difference.